pkg/test/virtualmachines.go

```diff
@@ -0,0 +1,78 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/karpenter-provider-azure/pkg/fake"
+	"github.com/imdario/mergo"
+	"github.com/samber/lo"
+)
+
+// VirtualMachineOptions customizes an Azure Virtual Machine for testing.
+type VirtualMachineOptions struct {
+	Name         string
+	NodepoolName string
```

Comment:
Having VM be aware of the Nodepool is a smell

Reply:
Ok, I see now this was maybe in response to my question
> What happens if options.Tags are not passed?

I was just asking about whether it would be tolerated / won't crash. Did not mean to suggest they should be calculated automatically (and definetely not at the expense of having to pass nodepool name). Will be more clear next time.